common/src/main/java/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -322,7 +324,13 @@ open class DeviceCache(
     @Synchronized
     fun getProductEntitlementMapping(): ProductEntitlementMapping? {
         return preferences.getString(productEntitlementMappingCacheKey, null)?.let { jsonString ->
-            return ProductEntitlementMapping.fromJson(JSONObject(jsonString))
+            return try {
+                ProductEntitlementMapping.fromJson(JSONObject(jsonString))
+            } catch (e: JSONException) {
+                errorLog(OfflineEntitlementsStrings.ERROR_PARSING_PRODUCT_ENTITLEMENT_MAPPING.format(jsonString), e)
+                preferences.edit().remove(productEntitlementMappingCacheKey).apply()
+                null
+            }
```

Comment:
Found this small edge case, without this, we would have crashed later when trying to get the cached mappings.